src/arrayMethodSort.js

```diff
@@ -4,8 +4,30 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (
+    compareFunction = (a, b) => {
+      if (a.toString() > b.toString()) {
+        return 1;
+      } else if (a.toString() < b.toString()) {
+        return -1;
+      } else {
+        return 0;
```

Comment:
The default compare function is converting elements to strings multiple times which is not efficient. Consider storing the string representations in variables before the comparison to avoid repeated conversions.